packages/@aws-cdk/cdk/lib/core/tokens.ts

```diff
@@ -29,13 +48,59 @@ export class Token {
 
         return value;
     }
+
+    /**
+     * Return a reversible string representation of this token
+     *
+     * If the Token is initialized with a literal, the stringified value of the
+     * literal is returned. Otherwise, a special quoted string representation
+     * of the Token is returned that can be embedded into other strings.
+     *
+     * Strings with quoted Tokens in them can be restored back into
+     * complex values with the Tokens restored by calling `resolve()`
+     * on the string.
+     */
+    public toString(): string {
+        const valueType = typeof this.valueOrFunction;
+        // Optimization: if we can immediately resolve this, don't bother
+        // registering a Token.
+        if (valueType === 'string' || valueType === 'number' || valueType === 'boolean') {
+            return this.valueOrFunction.toString();
+        }
+
+        if (this.tokenKey === undefined) {
+            this.tokenKey = TOKEN_STRING_MAP.register(this, this.displayName);
+        }
+        return this.tokenKey;
+    }
+
+    /**
+     * Turn this Token into JSON
+     *
+     * This gets called by JSON.stringify(). We want to prohibit this, because
+     * it's not possible to do this properly, so we just throw an error here.
+     */
+    public toJSON(): any {
+        throw new Error('JSON.stringify() cannot be applied to structure with a deferred Token in it. Use CloudFormationJSON.stringify() instead.');
+    }
+
+    /**
+     * Return a concated version of this Token in a string context
+     *
+     * The default implementation of this combines strings, but specialized
+     * implements of Token can return a more appropriate value.
+     */
```

Comment:
Does it make sense to provide a default? In what use case it will be invoked?

Reply:
It will be invoked when Tokenized string literals are being used (as in `new Token("literal")`).